### 1. Interpretation and significance

Suppose that we would like to know how much families in the US are spending on recreation annually. We've estimated the following model:

$$ expenditure = 873 + 0.0012annual\_income + 0.00002annual\_income^2 - 223.57have\_kids $$

*expenditure* is the annual spending on recreation in US dollars, *annual_income* is the annual income in US dollars, and *have_kids* is a dummy variable indicating the families with children. Interpret the estimated coefficients. What additional statistics should be given in order to make sure that your interpretations make sense statistically. Write up your answer.

### Answer 1: 

Assuming the coefficients are statistically significant (which we don't know because none of them have their associated statistics reported), we can say that the bias term is $873, the expenditure increases in quadratically with annual income, and families with kids spend 223.57 dollars fewer on recreation.

### 2. Weather model

In this exercise, you'll work with the historical temperature data from the previous checkpoint. To complete this assignment, submit a link to a Jupyter notebook containing your solutions to the following tasks:

* First, load the dataset from the **weatherinszeged** table from Thinkful's database.
* Build a linear regression model where your target variable is the difference between the *apparenttemperature* and the *temperature*. As explanatory variables, use *humidity* and *windspeed*. Now, estimate your model using OLS. Are the estimated coefficients statistically significant? Are the signs of the estimated coefficients in line with your previous expectations? Interpret the estimated coefficients. What are the relations between the target and the explanatory variables? 
* Next, include the interaction of *humidity* and *windspeed* to the model above and estimate the model using OLS. Are the coefficients statistically significant? Did the signs of the estimated coefficients for *humidity* and *windspeed* change? Interpret the estimated coefficients.

### Answer 2:

In [1]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

import statsmodels.api as sm
import statsmodels.formula.api as smf
from sqlalchemy import create_engine

# Display preferences.
%matplotlib inline
pd.options.display.float_format = "{:.3f}".format

%load_ext nb_black

import warnings

warnings.filterwarnings(action="ignore")

<IPython.core.display.Javascript object>

In [2]:
postgres_user = "dsbc_student"
postgres_pw = "7*.8G9QH21"
postgres_host = "142.93.121.174"
postgres_port = "5432"
postgres_db = "weatherinszeged"

engine = create_engine(
    "postgresql://{}:{}@{}:{}/{}".format(
        postgres_user, postgres_pw, postgres_host, postgres_port, postgres_db
    )
)
szeged_df = pd.read_sql_query("select * from weatherinszeged", con=engine)

# no need for an open connection, as we're only doing a single query
engine.dispose()


szeged_df.head(50)

date        summary preciptype  temperature  \
0  2006-03-31 22:00:00+00:00  Partly Cloudy       rain        9.472   
1  2006-03-31 23:00:00+00:00  Partly Cloudy       rain        9.356   
2  2006-04-01 00:00:00+00:00  Mostly Cloudy       rain        9.378   
3  2006-04-01 01:00:00+00:00  Partly Cloudy       rain        8.289   
4  2006-04-01 02:00:00+00:00  Mostly Cloudy       rain        8.756   
5  2006-04-01 03:00:00+00:00  Partly Cloudy       rain        9.222   
6  2006-04-01 04:00:00+00:00  Partly Cloudy       rain        7.733   
7  2006-04-01 05:00:00+00:00  Partly Cloudy       rain        8.772   
8  2006-04-01 06:00:00+00:00  Partly Cloudy       rain       10.822   
9  2006-04-01 07:00:00+00:00  Partly Cloudy       rain       13.772   
10 2006-04-01 08:00:00+00:00  Partly Cloudy       rain       16.017   
11 2006-04-01 09:00:00+00:00  Partly Cloudy       rain       17.144   
12 2006-04-01 10:00:00+00:00  Partly Cloudy       rain       17.800   
13 2006-04-01 11:00:00+00:00  Partly Cloudy       rain       17.333   
14 2006-04-01 12:00:00+00:00  Partly Cloudy       rain       18.878   
15 2006-04-01 13:00:00+00:00  Partly Cloudy       rain       18.911   
16 2006-04-01 14:00:00+00:00  Partly Cloudy       rain       15.389   
17 2006-04-01 15:00:00+00:00  Mostly Cloudy       rain       15.550   
18 2006-04-01 16:00:00+00:00  Mostly Cloudy       rain       14.256   
19 2006-04-01 17:00:00+00:00  Mostly Cloudy       rain       13.144   
20 2006-04-01 18:00:00+00:00  Mostly Cloudy       rain       11.550   
21 2006-04-01 19:00:00+00:00  Mostly Cloudy       rain       11.183   
22 2006-04-01 20:00:00+00:00  Partly Cloudy       rain       10.117   
23 2006-04-01 21:00:00+00:00  Mostly Cloudy       rain       10.200   
24 2006-04-09 22:00:00+00:00  Partly Cloudy       rain       10.422   
25 2006-04-09 23:00:00+00:00  Partly Cloudy       rain        9.911   
26 2006-04-10 00:00:00+00:00  Mostly Cloudy       rain       11.183   
27 2006-04-10 01:00:00+00:00  Partly Cloudy       rain        7.156   
28 2006-04-10 02:00:00+00:00  Partly Cloudy       rain        6.111   
29 2006-04-10 03:00:00+00:00  Partly Cloudy       rain        6.789   
30 2006-04-10 04:00:00+00:00  Mostly Cloudy       rain        7.261   
31 2006-04-10 05:00:00+00:00  Mostly Cloudy       rain        7.800   
32 2006-04-10 06:00:00+00:00  Mostly Cloudy       rain        9.872   
33 2006-04-10 07:00:00+00:00  Mostly Cloudy       rain       12.222   
34 2006-04-10 08:00:00+00:00  Mostly Cloudy       rain       15.094   
35 2006-04-10 09:00:00+00:00  Mostly Cloudy       rain       17.356   
36 2006-04-10 10:00:00+00:00  Mostly Cloudy       rain       19.006   
37 2006-04-10 11:00:00+00:00  Mostly Cloudy       rain       20.044   
38 2006-04-10 12:00:00+00:00  Mostly Cloudy       rain       21.050   
39 2006-04-10 13:00:00+00:00  Mostly Cloudy       rain       21.183   
40 2006-04-10 14:00:00+00:00  Mostly Cloudy       rain       20.117   
41 2006-04-10 15:00:00+00:00  Mostly Cloudy       rain       20.217   
42 2006-04-10 16:00:00+00:00  Mostly Cloudy       rain       20.000   
43 2006-04-10 17:00:00+00:00  Mostly Cloudy       rain       17.800   
44 2006-04-10 18:00:00+00:00  Mostly Cloudy       rain       16.061   
45 2006-04-10 19:00:00+00:00  Mostly Cloudy       rain       15.022   
46 2006-04-10 20:00:00+00:00       Overcast       rain       14.422   
47 2006-04-10 21:00:00+00:00       Overcast       rain       14.256   
48 2006-04-10 22:00:00+00:00       Overcast       rain       13.772   
49 2006-04-10 23:00:00+00:00       Overcast       rain       13.283   

    apparenttemperature  humidity  windspeed  windbearing  visibility  \
0                 7.389     0.890     14.120      251.000      15.826   
1                 7.228     0.860     14.265      259.000      15.826   
2                 9.378     0.890      3.928      204.000      14.957   
3                 5.944     0.830     14.104      269.000      15.826   
4                 6.978     0.830    

<IPython.core.display.Javascript object>

In [3]:
szeged_df.isna().mean()

date                  0.000
summary               0.000
preciptype            0.000
temperature           0.000
apparenttemperature   0.000
humidity              0.000
windspeed             0.000
windbearing           0.000
visibility            0.000
loudcover             0.000
pressure              0.000
dailysummary          0.000
dtype: float64

<IPython.core.display.Javascript object>

In [4]:
szeged_df.describe()

temperature  apparenttemperature  humidity  windspeed  windbearing  \
count    96453.000            96453.000 96453.000  96453.000    96453.000   
mean        11.933               10.855     0.735     10.811      187.509   
std          9.552               10.697     0.195      6.914      107.383   
min        -21.822              -27.717     0.000      0.000        0.000   
25%          4.689                2.311     0.600      5.828      116.000   
50%         12.000               12.000     0.780      9.966      180.000   
75%         18.839               18.839     0.890     14.136      290.000   
max         39.906               39.344     1.000     63.853      359.000   

       visibility  loudcover  pressure  
count   96453.000  96453.000 96453.000  
mean       10.347      0.000  1003.236  
std         4.192      0.000   116.970  
min         0.000      0.000     0.000  
25%         8.340      0.000  1011.900  
50%        10.046      0.000  1016.450  
75%        14.812      0.000  1021.090  
max        16.100      0.000  1046.380

<IPython.core.display.Javascript object>

In [15]:
szeged_df["target_temp"] = szeged_df["apparenttemperature"] - szeged_df["temperature"]

<IPython.core.display.Javascript object>

In [16]:
szeged_df.describe()

temperature  apparenttemperature  humidity  windspeed  windbearing  \
count    96453.000            96453.000 96453.000  96453.000    96453.000   
mean        11.933               10.855     0.735     10.811      187.509   
std          9.552               10.697     0.195      6.914      107.383   
min        -21.822              -27.717     0.000      0.000        0.000   
25%          4.689                2.311     0.600      5.828      116.000   
50%         12.000               12.000     0.780      9.966      180.000   
75%         18.839               18.839     0.890     14.136      290.000   
max         39.906               39.344     1.000     63.853      359.000   

       visibility  loudcover  pressure  target_temp  
count   96453.000  96453.000 96453.000    96453.000  
mean       10.347      0.000  1003.236       -1.078  
std         4.192      0.000   116.970        1.679  
min         0.000      0.000     0.000      -10.183  
25%         8.340      0.000  1011.900       -2.217  
50%        10.046      0.000  1016.450        0.000  
75%        14.812      0.000  1021.090        0.000  
max        16.100      0.000  1046.380        4.811

<IPython.core.display.Javascript object>

In [17]:
y = szeged_df["target_temp"]
X = szeged_df[["humidity", "windspeed"]]

<IPython.core.display.Javascript object>

In [18]:
X = sm.add_constant(X)

results = sm.OLS(y, X).fit()

results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            target_temp   R-squared:                       0.288
Model:                            OLS   Adj. R-squared:                  0.288
Method:                 Least Squares   F-statistic:                 1.949e+04
Date:                Tue, 07 Apr 2020   Prob (F-statistic):               0.00
Time:                        20:37:48   Log-Likelihood:            -1.7046e+05
No. Observations:               96453   AIC:                         3.409e+05
Df Residuals:                   96450   BIC:                         3.409e+05
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          2.4381      0.021    115.948      0.000       2.397       2.479
humidity      -3.0292      0.024   -126.479      0.000      -3.076      -2.982
windspeed     -0.1193      0.001   -176.164      0.000      -0.121      -0.118
==============================================================================
Omnibus:                     3935.747   Durbin-Watson:                   0.267
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             4613.311
Skew:                          -0.478   Prob(JB):                         0.00
Kurtosis:                       3.484   Cond. No.                         88.1
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

<IPython.core.display.Javascript object>

The coefficients are statistically significant at p < 0.001.  The difference in apparent versus regular temperature (wind chill?), i.e. target_temp, seems to decrease with increase in humidity or windspeed.  For a one unit increase in humidity, the target_temp decreases by 3.03 degrees.  For a one unit increase in windspeed, the target_temp decreases by 0.012 degrees.

In [19]:
szeged_df["wind_hum_interaction"] = szeged_df["windspeed"] * szeged_df["humidity"]

<IPython.core.display.Javascript object>

In [20]:
y = szeged_df["target_temp"]
X2 = szeged_df[["humidity", "windspeed", "wind_hum_interaction"]]

<IPython.core.display.Javascript object>

In [21]:
X2 = sm.add_constant(X2)

results = sm.OLS(y, X2).fit()

results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            target_temp   R-squared:                       0.341
Model:                            OLS   Adj. R-squared:                  0.341
Method:                 Least Squares   F-statistic:                 1.666e+04
Date:                Tue, 07 Apr 2020   Prob (F-statistic):               0.00
Time:                        20:55:17   Log-Likelihood:            -1.6669e+05
No. Observations:               96453   AIC:                         3.334e+05
Df Residuals:                   96449   BIC:                         3.334e+05
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
========================================================================================
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
const                    0.0839      0.033      2.511      0.012       0.018       0.149
humidity                 0.1775      0.043      4.133      0.000       0.093       0.262
windspeed                0.0905      0.002     36.797      0.000       0.086       0.095
wind_hum_interaction    -0.2971      0.003    -88.470      0.000      -0.304      -0.291
==============================================================================
Omnibus:                     4849.937   Durbin-Watson:                   0.265
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             9295.404
Skew:                          -0.378   Prob(JB):                         0.00
Kurtosis:                       4.320   Cond. No.                         193.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

<IPython.core.display.Javascript object>

All of the coefficients are still statistically significant at the p < 0.01 level.  

The signs of the humidity and windspeed coefficients are now positive.  The interaction variable has a negative sign and a larger magnitude than either of the individual coefficients.

For a given windspeed, a 1 point increase in humidity has a .17-0.3X windspeed point change in the target variable.  For a given humidity, a one point increase in windspeed results in a 0.09-0.3X humidity point change in the target.

###  3. House prices model

In this exercise, you'll interpret your house prices model. To complete this assignment, submit a link to a Jupyter notebook containing your solutions to the following tasks:

* Load the **houseprices** data from Thinkful's database.
* Run your house prices model again and interpret the results. Which features are statistically significant, and which are not?
* Now, exclude the insignificant features from your model. Did anything change?
* Interpret the statistically significant coefficients by quantifying their relations with the house prices. Which features have a more prominent effect on house prices?
* Do the results sound reasonable to you? If not, try to explain the potential reasons.

### Answer 3:

Will do when I finish the houseprices assignment.  TBD.